From 4b2ed58b0d467cc2fd184524c4aa184554581e34 Mon Sep 17 00:00:00 2001
From: dramforever <dramforever@live.com>
Date: Thu, 24 Jul 2025 04:26:40 +0800
Subject: [PATCH] wayback-session: Use &argv[1] not &argv[optind]

With common/optparse.c, optind is no longer from libc, and doesn't start
at 1. Since we don't parse arguments (yet) just use &argv[1] instead.

Closes #56
---
 wayback-session/wayback-session.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wayback-session/wayback-session.c b/wayback-session/wayback-session.c
index e4834f0..c0b2356 100644
--- a/wayback-session/wayback-session.c
+++ b/wayback-session/wayback-session.c
@@ -64,7 +64,7 @@ int main(int argc, char *argv[])
 	if (argc == 1) {
 		xinitrc_path = get_xinitrc_path();
 	} else {
-		session_cmd = &argv[optind];
+		session_cmd = &argv[1];
 	}
 
 	char *xwayback_path = getenv("XWAYBACK_PATH");
-- 
GitLab

